baseplate/live_data/watcher.py

```diff
@@ -0,0 +1,106 @@
+"""Watch nodes in ZooKeeper and sync their contents to disk on change."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import errno
+import logging
+import os
+import sys
+import time
+
+from .._compat import configparser
+from .. import config
+from ..secrets import secrets_store_from_config
+from .zookeeper import zookeeper_client_from_config
+
+
+logger = logging.getLogger(__name__)
+
+
+HEARTBEAT_INTERVAL = 60
+
+
+class NodeWatcher(object):
+    def __init__(self, source, dest):
+        self.source = source
+        self.dest = dest
+
+    def on_change(self, data, _):
+        if data is None:
+            # the data node does not exist
+            try:
+                logger.info("Removing %r; watched node deleted.", self.dest)
+                os.unlink(self.dest)
+            except OSError as exc:
+                logger.debug("%s: couldn't unlink: %s", self.dest, exc)
+            return
+
+        logger.info("Updating %r", self.dest)
+        with open(self.dest + ".tmp", "wb") as tmpfile:
```

Comment:
Yeah. I was just assuming that whatever could be here would be a secret and so we'd want to make sure no one is snooping. Also wanted to be wary of naming attacks since the filename is static.